pkg/genericclioptions/genericclioptions.go

```diff
@@ -23,13 +28,120 @@ type IOStreams struct {
 	ErrOut io.Writer
 }
 
-type ClientConfig struct {
-	Kubeconfig    string
+type sharedClientConfig struct {
 	QPS           float32
 	Burst         int
 	UserAgentName string
-	RestConfig    *restclient.Config
-	ProtoConfig   *restclient.Config
+}
+
+func newSharedClientConfig(userAgentName string) sharedClientConfig {
+	return sharedClientConfig{
+		QPS:           50,
+		Burst:         75,
+		UserAgentName: userAgentName,
+	}
+}
+
+func (scc *sharedClientConfig) AddFlags(cmd *cobra.Command) {
+	cmd.PersistentFlags().Float32VarP(&scc.QPS, "qps", "", scc.QPS, "Maximum allowed number of queries per second.")
+	cmd.PersistentFlags().IntVarP(&scc.Burst, "burst", "", scc.Burst, "Allows extra queries to accumulate when a client is exceeding its rate.")
+}
+
+func (scc *sharedClientConfig) Validate() error {
+	return nil
+}
+
+func (scc *sharedClientConfig) Complete() error {
+	return nil
+}
+
+type ClientConfigSet struct {
+	sharedClientConfig
+	Kubeconfigs   []string
+	ClientConfigs []ClientConfig
```

Comment:
ClientConfig  already contains a kubeconfig - this seems to be duplicit

Reply:
this is there to keep the flags as they are now - iirc it was a requirement that we don't need to change the existing CI jobs

Reply:
done